lib/openxml/builder.rb

```diff
@@ -28,39 +28,24 @@ def to_s
     end
     alias :to_xml :to_s
 
-    # Adapted from Nokogiri's builder.rb
     def [](ns)
       if @parent != @document
-        @ns = @parent.namespace_definitions.find { |x| x.prefix == ns.to_s }
-      end
-      return self if @ns
-
-      @parent.ancestors.each do |a|
-        next if a == @document
-        @ns = a.namespace_definitions.find { |x| x.prefix == ns.to_s }
+        @ns = @parent.namespaces.find { |namespace| namespace.prefix == ns.to_s }
         return self if @ns
       end
 
-      @ns = { :pending => ns.to_s }
+      @ns = OpenXml::Builder::Namespace.new(ns.to_s, nil)
```

Comment:
It's not valid without a URI, but in order to have a valid URI, we'd have to use one of the workarounds we used previously to do validation (traversing all ancestors OR passing defined namespaces on to children). Since we're skipping validation, perhaps this doesn't need to be its own class anymore and could suffice to just be a symbol. I'll make that refactor.